src/main/java/com/google/javascript/gents/TypeAnnotationPass.java

```diff
@@ -199,14 +199,19 @@ public void visit(NodeTraversal t, Node n, Node parent) {
           // creating each MEMBER_VARIABLE_DEFs so code generator works.
           if (bestJSDocInfo != null && bestJSDocInfo.hasTypedefType()) {
             Node typedefTypeRoot = bestJSDocInfo.getTypedefType().getRoot();
-            //LC
-            //    LB
-            //        COLON
-            //            STRING_KEY a
-            //            STRING number
-            //        COLON
-            //            STRING_KEY b
-            //            STRING number
+            if ((typedefTypeRoot.getToken() == Token.BANG)
+                || (typedefTypeRoot.getToken() == Token.QMARK)) {
+              typedefTypeRoot = typedefTypeRoot.getFirstChild();
+            }
+            //(BANG|QMASK)
+            //    LC
+            //        LB
```

Comment:
Yeah this invariant doesn't hold if the input is a class inner typedef with a simple type alias.

To fix this, I think it doesn't make sense to convert a simple alias to a TypeScript interface -  it should be a type alias instead. We should only create an interface if it's an inner typedef and it contains an object. WDYT?